app/controllers/concerns/incomplete_items_concern.rb

```diff
@@ -0,0 +1,37 @@
+module IncompleteItemsConcern
+  extend ActiveSupport::Concern
+
+  private
+
+  def incomplete_items_summary
+    @incomplete_items_summary ||= Nsm::IncompleteItems.new(current_application, item_type,
+                                                           self)
+  end
+
+  def build_items_incomplete_flash
+    flash = if import_notification_needed?
```

Comment:
No need for this to be a variable

Reply:
it needs to be a variable because: 

- the function needs to return the flash hash
- the function needs to set the "viewed" flag to true only after the hash has been generated fgrom the if statements, since import_notification_needed? relies on that flag to check if the page has been viewed or not 

Therefore I need to pre-store the flash to ensure it follows the correct path, reset the value, then return it

Reply:
I think this might be better reviewed on a call